### PR TITLE
Matched Update_camera_mode_4/6

### PR DESCRIPTION
**Pull request by MS-DOS1999**

### CODE REVIEW

**Comment by pixel-stuck** [comment]
Looks like you should probably run coverage.py and tell it yes when it asks to delete the ASM files

File: src/9d10_len_1080.c [hunk 1]
```diff
@@ -1,7 +1,106 @@
...
+        camera->lookAt_eye.y = camera->lookAt_obj.y + deltaY2;
+        camera->lookAt_eye.z = camera->lookAt_obj.z + deltaZ2;
+    }
+    camera->lookAt_obj.x = camera->lookAt_obj_target.x;
+    camera->lookAt_obj.y = camera->lookAt_obj_target.y + camera->currentYOffset;
+    camera->lookAt_obj.z = camera->lookAt_obj_target.z;
+    boomYaw = DEG_TO_RAD(camera->currentBoomYaw);
+    sinBoom = sin_rad(boomYaw);
+    cosBoom = cos_rad(boomYaw);
+    deltaY = (deltaX = 0.0f);
```
Comment (pixel-stuck) [inline src/9d10_len_1080.c:78]: parents can probably be removed here

File: src/9d10_len_1080.c [hunk 2]
```diff
@@ -1,7 +1,106 @@
...
+    deltaZ = camera->currentBoomLength;
+    deltaX2 = deltaX;
+    deltaY2 = deltaY;
+    boomYaw = (deltaX = -deltaY2);
```
Comment (pixel-stuck) [inline src/9d10_len_1080.c:82]: here too

File: src/9d10_len_1080.c [hunk 3]
```diff
@@ -1,7 +1,106 @@
...
+    deltaZ2 = sinBoom * deltaX + deltaZ * cosBoom;
+    camera->lookAt_eye.x = camera->lookAt_obj.x + deltaX2;
+    camera->lookAt_eye.y = camera->lookAt_obj.y + deltaY2;
+    camera->lookAt_eye.z = camera->lookAt_obj.z + deltaZ2;
+    camera->currentYaw = atan2(camera->lookAt_eye.x, camera->lookAt_eye.z, camera->lookAt_obj.x, camera->lookAt_obj.z);
+    deltaX = camera->lookAt_obj.x - camera->lookAt_eye.x;
+    deltaY = camera->lookAt_obj.y - camera->lookAt_eye.y;
+    deltaZ = camera->lookAt_obj.z - camera->lookAt_eye.z;
+    camera->currentBlendedYawNegated = -atan2(0.0f, 0.0f, deltaX, deltaZ);
+    camera->currentPitch = atan2(0.0f, 0.0f, deltaY, -sqrtf((deltaX * deltaX) + (deltaZ * deltaZ)));
```
Comment (pixel-stuck) [inline src/9d10_len_1080.c:102]: SQ macro for squaring deltaX and deltaZ

**Comment by BowserSlug** [comment]
🚀🚀 This PR matches 2 functions (+0.05%) on `us`.

File: src/ad90_len_2880.c [hunk 4]
```diff
@@ -31,41 +29,45 @@ void update_camera_mode_6(Camera* camera) {
...
-        temp_f24_2 = (sinBoomYaw * -0.0f) + (camera->currentBoomLength * cosBoomYaw);
-        trueRotX = DEG_TO_RAD(camera->trueRotation.x);
-        sinRotX = sin_rad(trueRotX);
-        cosRotX = cos_rad(trueRotX);
-        camera->lookAt_eye.x = camera->lookAt_obj.x + ((cosRotX * 0.0f) - (temp_f24_2 * sinRotX));
-        camera->lookAt_eye.y = camera->lookAt_obj.y + temp_f30;
-        camera->lookAt_eye.z = camera->lookAt_obj.z + ((sinRotX * 0.0f) + (temp_f24_2 * cosRotX));
+        sinBoom = sin_rad(boomYaw);
+        cosBoom = cos_rad(boomYaw);
+        deltaY = (deltaX = 0.0f);
```
Comment (pixel-stuck) [inline src/ad90_len_2880.c:43]: Similar here for the parens

File: src/ad90_len_2880.c [hunk 5]
```diff
@@ -31,41 +29,45 @@ void update_camera_mode_6(Camera* camera) {
...
+        deltaZ = camera->currentBoomLength;
+        deltaX2 = deltaX;
+        deltaY2 = deltaY;
+        boomYaw = (deltaX = -deltaY2);
```
Comment (pixel-stuck) [inline src/ad90_len_2880.c:47]: Parents again

File: src/ad90_len_2880.c [hunk 6]
```diff
@@ -31,41 +29,45 @@ void update_camera_mode_6(Camera* camera) {
...
-    xDiff = camera->lookAt_obj.x - camera->lookAt_eye.x;
-    yDiff = camera->lookAt_obj.y - camera->lookAt_eye.y;
-    zDiff = camera->lookAt_obj.z - camera->lookAt_eye.z;
-    camera->currentBlendedYawNegated = -atan2(0.0f, 0.0f, xDiff, zDiff);
-    camera->currentPitch = atan2(0.0f, 0.0f, yDiff, -sqrtf((xDiff * xDiff) + (zDiff * zDiff)));
+    deltaX = camera->lookAt_obj.x - camera->lookAt_eye.x;
+    deltaY = camera->lookAt_obj.y - camera->lookAt_eye.y;
+    deltaZ = camera->lookAt_obj.z - camera->lookAt_eye.z;
+    camera->currentBlendedYawNegated = -atan2(0.0f, 0.0f, deltaX, deltaZ);
+    camera->currentPitch = atan2(0.0f, 0.0f, deltaY, -sqrtf((deltaX * deltaX) + (deltaZ * deltaZ)));
```
Comment (pixel-stuck) [inline src/ad90_len_2880.c:69]: SQ macro

File: src/ad90_len_2880.c [hunk 7]
```diff
@@ -1,23 +1,21 @@
...
-    f32 trueRotX;
-    f32 sinRotX;
-    f32 cosRotX;
-    f32 temp_f24_2;
-    f32 temp_f30;
-    s32 var_v1;
+    f32 new_var;
 
-    if (camera->unk_06 != 0 || camera->changingMap) {
+    if (camera->unk_06 != 0 || camera->isChangingMap != 0) {
```
Comment (ethteck) [inline src/ad90_len_2880.c:16]:
```suggestion
    if (camera->unk_06 != 0 || camera->isChangingMap) {
```

File: src/ad90_len_2880.c [hunk 8]
```diff
@@ -1,23 +1,21 @@
...
         camera->unk_06 = 0;
-        camera->changingMap = 0;
+        camera->isChangingMap = 0;
```
Comment (ethteck) [inline src/ad90_len_2880.c:18]:
```suggestion
        camera->isChangingMap = FALSE;
```

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/9d10_len_1080.c [hunk 9]
```diff
@@ -1,7 +1,106 @@
...
+        camera->unk_9C = 0.0f;
+        camera->lookAt_obj.x = camera->lookAt_obj_target.x;
+        camera->lookAt_obj.y = camera->lookAt_obj_target.y + camera->currentYOffset;
+        camera->lookAt_obj.z = camera->lookAt_obj_target.z;
+        boomYaw = DEG_TO_RAD(camera->currentBoomYaw);
+        sinBoom = sin_rad(boomYaw);
+        cosBoom = cos_rad(boomYaw);
+        deltaY = deltaX = 0.0f;
+        deltaZ = camera->currentBoomLength;
+        deltaX2 = deltaX;
```
Comment (HailSanta) [inline src/9d10_len_1080.c:54]: Does it still match if you set these to 0.0 instead of chaining them? It would be cleaner if so.

Comment (ethteck) [reply src/9d10_len_1080.c:54]: yee, 
x = 0
y = 0 would look nicer imo

**Review by nanaian — APPROVED** [review]

**Pull request merged**